### PR TITLE
#529: [KitchenSink] Default style for components (closes #529)

### LINKED ISSUE #529: [KitchenSink] Default style for components
← #490

## requirements
Define default style and overridable style variables for each component, following guidelines.

## specs
Guidelines: [https://zpl.io/Z1jerU](https://zpl.io/Z1jerU)

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by veej**
Issue #529

#### Badge
image

#### Button
image image image image 

#### Dropdown
image

#### DropdownMenu
image

#### Modal
image

#### MoreOrLess
image

#### Panel
image

#### TabbedPanel
image

#### Table
image

#### Tooltip
image

### CODE REVIEW

File: src/button/button.scss [hunk 1]
```diff
@@ -1,64 +1,66 @@
...
-$border-color: black !default;
+$font-weight: $semibold !default;
+$background: $gradientGrey !default;
+$color: $darkGreyBlue !default;
+$border-color: $silver !default;
 $border-width: 1px !default;
-$border-radius: 3px !default;
-$font-family: sans-serif !default;
+$border-radius: 4px !default;
+$font-family: "Source Sans Pro", sans-serif !default;
```
Comment (FrancescoCioria) [inline src/button/button.scss]: IMO brc shouldn't set `font-family` as it's inherited by default --> it's one of those CSS properties you set once in your app to `body` or `html`. @veej do you agree?

File: src/confirmation-input/confirmationInput.scss [hunk 2]
```diff
@@ -1,18 +1,40 @@
...
-$color: lightblue !default;
-$hover-color: blue !default;
+$font-weight: $semibold !default;
+$color: $azure !default;
+$hover-color: $azure !default;
 $disabled-opacity: .6 !default;
 
 .confirmation-input {
 
+  font-family: "Source Sans Pro", sans-serif;
```
Comment (FrancescoCioria) [inline src/confirmation-input/confirmationInput.scss]: same here

File: src/confirmation-input/confirmationInput.scss [hunk 3]
```diff
@@ -1,18 +1,40 @@
+@import '../utils/theme/variables.scss';
+
 $padding-h: 12px !default;
 $font-size: 14px !default;
-$font-weight: 500 !default;
-$color: lightblue !default;
-$hover-color: blue !default;
+$font-weight: $semibold !default;
+$color: $azure !default;
+$hover-color: $azure !default;
```
Comment (FrancescoCioria) [inline src/confirmation-input/confirmationInput.scss]: please rename to `color-hover`

File: src/dropdown/dropdown.scss [hunk 4]
```diff
@@ -300,17 +301,17 @@ $arrow-color: $text-color !default;
 $arrow-color-hover: $text-color-hover !default;
 $arrow-color-open: $text-color-open !default;
 
-$x-color: darkgray !default;
-$x-color-hover: gray !default;
+$x-color: $text-color !default;
+$x-color-hover: $text-color-hover !default;
 
 $menu-background: white !default;
```
Comment (FrancescoCioria) [inline src/dropdown/dropdown.scss]: shouldn't this use `$white`?

File: src/DropdownMenu/dropdownMenu.scss [hunk 5]
```diff
@@ -1,8 +1,29 @@
+@import '../utils/theme/variables.scss';
+
+$text-color: $coolGrey;
```
Comment (FrancescoCioria) [inline src/DropdownMenu/dropdownMenu.scss]: I think you forgot to add `!default` to these variables

File: src/modal/modal.scss [hunk 6]
```diff
@@ -1,21 +1,24 @@
+@import '../utils/theme/variables.scss';
+
```
Comment (FrancescoCioria) [inline src/modal/modal.scss]: why didn't you add the API to this sass? will you do it in another issue?

File: src/Panel/panel.scss [hunk 7]
```diff
@@ -1,16 +1,19 @@
-$border-color: #e6e6e6 !default;
+@import '../utils/theme/variables.scss';
+
+$border-color: $silver !default;
 $soft-loader-color: #45a7b9 !default;
 
-$header-background: white !default;
+$header-background: $gradientGrey !default;
 $header-background-dark: #2c3a40 !default;
```
Comment (FrancescoCioria) [inline src/Panel/panel.scss]: no variable?

File: src/Panel/panel.scss [hunk 8]
```diff
@@ -1,16 +1,19 @@
-$border-color: #e6e6e6 !default;
+@import '../utils/theme/variables.scss';
+
+$border-color: $silver !default;
 $soft-loader-color: #45a7b9 !default;
```
Comment (FrancescoCioria) [inline src/Panel/panel.scss]: no variable?

File: src/Panel/panel.scss [hunk 9]
```diff
@@ -1,16 +1,19 @@
...
-$expand-icon-color: white !default;
-$expand-icon-color-hover: rgba(white, 0.7) !default;
+$expand-icon-color: $coolGrey !default;
+$expand-icon-color-hover: rgba($coolGrey, 0.7) !default;
 
-$content-background: white !default;
-$content-color: #2c3a40 !default;
-$content-title-color: #2c3a40 !default;
+$content-background: $white !default;
+$content-color: #313D4F !default;
```
Comment (FrancescoCioria) [inline src/Panel/panel.scss]: no variable?

File: src/Panel/panel.scss [hunk 10]
```diff
@@ -1,16 +1,19 @@
...
+$content-title-color: #313D4F !default;
```
Comment (FrancescoCioria) [inline src/Panel/panel.scss]: no variable?

File: src/Panel/panel.scss [hunk 11]
```diff
@@ -107,11 +107,7 @@ $content-title-color: #313D4F !default;
 
     .panel-header-title {
       font-size: 16px;
-      font-weight: 700;
-    }
-
-    .panel-header-content {
-      margin-left: 5px;
```
Comment (FrancescoCioria) [inline src/Panel/panel.scss]: why did you remove the `margin-left`? has it changed in the guidelines?

File: src/button/button.scss (hunk 1 above)
Comment (veej) [inline src/button/button.scss]: Yep... actually I came to the same conclusion, but I forgot to remove it

File: src/collapsable-section/collapsableSection.scss [hunk 12]
```diff
@@ -1,32 +1,40 @@
...
 $background: transparent !default;
 
-$content-background: transparent !default;
-$is-open-content-background: transparent !default;
-$is-selected-content-background: transparent !default;
-$hover-content-background: transparent !default;
+$header-background: $gradientGrey !default;
+$is-open-header-background: $gradientGrey !default;
+$is-selected-header-background: transparent !default;
+$hover-header-background: $gradientDarkGrey !default;
```
Comment (FrancescoCioria) [inline src/collapsable-section/collapsableSection.scss]: some of these variables should be renamed:
```
$is-open-header-background --> $header-background-open
$is-selected-header-background --> $header-background-selected
$hover-header-background --> $header-background-hover
```

File: src/Panel/panel.scss (hunk 11 above)
Comment (veej) [inline src/Panel/panel.scss]: As discussed offline, it's hard to split pure style/color changes from structural changes. Even if we want to add or rename SASS variables, they should be adapted in projects that use them. 
IMO, if we want to keep this new style as the default for buildo components, this PR should become `breaking` and we have to evaluate/adapt every structural change in projects that use those components. 
@giogonzo, any thoughts?

Comment (giogonzo) [inline src/Panel/panel.scss]: yes, +1

Comment (FrancescoCioria) [inline src/Panel/panel.scss]: ok, then I'll merge this!
@veej can you open on AlinIQ and IperCron an issue to fix the style of the components you've touched (please list them all in the issue and point to this PR so we don't forget the changes)
